k8s-subst

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# This script generates K8s yaml files from a set of templates.  To run it you need to 
+# define env vars IMAGE_PATH and IMAGE_TAG such that clearwater images can be found at
+#   IMAGE_PATH/{container name e.g. bono}:IMAGE_TAG
+# The script takes a single parameter which is the path to the Kubernetes directory 
+# (which has the templates subdirectory).
+
+[ -z "$IMAGE_PATH" ] && { echo "Need to set IMAGE_PATH"; exit 1; }
+[ -z "$IMAGE_TAG" ] && { echo "Need to set IMAGE_TAG"; exit 1; }
+[ -z "$1" ] && { echo "Need to specify path to K8s tmpl files"; exit 1; }
+
+TMPL_PATH=$1/templates/*
+
+for f in $TMPL_PATH
+do
+  filename=${f##*/}
+  containername="${filename%.*}" 
+  sed 's@{{IMAGE:\([^}]*\)}}@'$IMAGE_PATH'\/\1\:'$IMAGE_TAG'@g' $1/templates/$containername.tmpl > $1/$containername.yaml
```

Comment:
Coming back to this after looking below, i think it would be much nicer to do this in separate steps.
in the depl files you've got {{IMAGE NAME:<process>}}, and you've got some messy regex stuff here to pull that out, and include it.
Could we not do this by having the template laid out as `{{IMAGE NAME}}:<process>{{IMAGE TAG}}`, and then in this script:

* Copy .tmpl out to .yaml
* Sed to replace {{IMAGE NAME}}
* Sed to replace/remove {{IMAGE TAG}}

possibly we could simplify the filename bit as well, but i can't work out quite how, and i'm not too worried if we can split the sed command, and add some comments on what each step is doing in a human readable form.